src/sip-call.ts

```diff
@@ -0,0 +1,297 @@
+  sections: string[],
+  mediaType: 'audio'
+): RtpStreamOptions {
+  try {
+    const section = sections.find((s) => s.startsWith('m=' + mediaType))
+
+    const { port } = sdp.parseMLine(section),
+    lines: string[] = sdp.splitLines(section),
+    rtcpLine = lines.find((l: string) => l.startsWith('a=rtcp:')),
+    ssrcLine = lines.find((l: string) => l.startsWith('a=ssrc'))
+
+    return {
+      port,
+      rtcpPort: (rtcpLine && Number(rtcpLine.match(/rtcp:(\S*)/)?.[1])) || port + 1, // if there is no explicit RTCP port, then use RTP port + 1
+      ssrc:     (ssrcLine && Number(ssrcLine.match(/ssrc:(\S*)/)?.[1])) || undefined
+    }
+  } catch (e) {
+    log.error('Failed to parse SDP from remote end')
+    log.error(sections.join('\r\n'))
+    throw e
+  }
+}
+
+function parseRtpDescription(log: Logger, inviteResponse: {
+  content: string
+}): RtpDescription {
+  const sections: string[] = sdp.splitSections(inviteResponse.content),
+    lines: string[] = sdp.splitLines(sections[0]),
+    cLine = lines.find((line: string) => line.startsWith('c='))!
+
+  return {
+    address: cLine.match(/c=IN IP4 (\S*)/)![1],
+    audio: getRtpDescription(log, sections, 'audio')
+  }
+}
+
+export class SipCall {
+  private seq = 20
+  private fromParams = { tag: getRandomId() }
+  private toParams: { tag?: string } = {}
+  private callId = getRandomId()
+  private sipStack: SipStack
+  public onEndedByRemote?: () => void
+  private destroyed = false
+  private readonly log: Logger
+
+  constructor(
+    log: Logger,
+    private sipOptions: SipOptions,
+  ) {
+    this.log = log;
+    const fromUri = sip.parseUri(this.sipOptions.from);
+    const host = fromUri.host;
+    const port = fromUri.port;
+
+    const { from } = this.sipOptions
+    this.sipStack = {
+      makeResponse: sip.makeResponse, 
+      ...sip.create({
+        host,
+        hostname: host,
+        port: port,
+        udp: true,
+        tcp: false,
+        tls: false,
+        ws: false
+      },
+      (request: SipRequest) => {
+        if (request.method === 'BYE') {
+          this.log.info('received BYE from remote end')
+          this.sipStack.send(this.sipStack.makeResponse(request, 200, 'Ok'))
+
+          if (this.onEndedByRemote) {
+            this.onEndedByRemote()
+          }
+        } 
+      }
+    )}
+  }
+
+  request({
+    method,
+    headers,
+    content,
+    seq,
+  }: {
+    method: string
+    headers?: Partial<SipHeaders>
+    content?: string
+    seq?: number
+  }) {
+    if (this.destroyed) {
+      return Promise.reject(
+        new Error('SIP request made after call was destroyed')
+      )
+    }
+
+    return new Promise<SipResponse>((resolve, reject) => {
+      seq = seq || this.seq++
+      this.sipStack.send(
+        {
+          method,
+          uri: this.sipOptions.to,
+          headers: {
+            to: {
+              name: '"SIP doorbell client"',
+              uri: this.sipOptions.to,
+              params: this.toParams,
```

Comment:
Fixes issue with my 2N SIP doorbell not correctly responding to the INVITE if it receives an empty `tag` in the `to:` header.

```suggestion
              params: this.toParams.tag ? this.toParams : undefined,
```